File: yuairun-backend/app/models/knowledge.py
"""知识库相关的 Pydantic 数据模型"""

from datetime import datetime
from typing import Optional

from pydantic import BaseModel, Field


# ============================================================
# 知识库
# ============================================================

class KnowledgeBaseInfo(BaseModel):
    """知识库信息"""
    id: str = Field(description="知识库ID")
    name: str = Field(description="知识库名称")
    description: str = Field(default="", description="知识库描述")
    isSystem: bool = Field(default=False, validation_alias="is_system", description="是否为系统内置")
    docCount: int = Field(default=0, validation_alias="doc_count", description="文档数量")
    chunkCount: int = Field(default=0, validation_alias="chunk_count", description="分块数量")
    createdAt: datetime = Field(validation_alias="created_at", description="创建时间")
    updatedAt: datetime = Field(validation_alias="updated_at", description="更新时间")


class CreateKnowledgeBaseRequest(BaseModel):
    """创建知识库请求"""
    name: str = Field(min_length=1, max_length=128, description="知识库名称")
    description: str = Field(default="", max_length=512, description="知识库描述")


class KnowledgeBaseListResponse(BaseModel):
    """知识库列表响应"""
    success: bool = True
    data: list[KnowledgeBaseInfo] = Field(default=[])
    error: str | None = None


class KnowledgeBaseDeleteResponse(BaseModel):
    """删除知识库响应"""
    success: bool = True
    error: str | None = None


# ============================================================
# 文档
# ============================================================

class KnowledgeDocumentInfo(BaseModel):
    """文档信息"""
    id: str = Field(description="文档ID")
    kbId: str = Field(validation_alias="kb_id", description="所属知识库ID")
    filename: str = Field(description="原始文件名")
    fileType: str = Field(validation_alias="file_type", description="文件类型")
    fileSize: int = Field(default=0, validation_alias="file_size", description="文件大小")
    pageCount: int | None = Field(default=None, validation_alias="page_count", description="页数")
    charCount: int = Field(default=0, validation_alias="char_count", description="字符数")
    chunkCount: int = Field(default=0, validation_alias="chunk_count", description="分块数")
    status: str = Field(default="pending", description="状态")
    errorMsg: str | None = Field(default=None, validation_alias="error_msg", description="错误信息")
    createdAt: datetime = Field(validation_alias="created_at", description="上传时间")


class KnowledgeDocumentListResponse(BaseModel):
    """文档列表响应"""
    success: bool = True
    data: list[KnowledgeDocumentInfo] = Field(default=[])
    error: str | None = None


class DocumentUploadResponse(BaseModel):
    """文档上传响应"""
    success: bool = True
    documentId: str = Field(validation_alias="document_id", description="文档ID")
    error: str | None = None


class DocumentDeleteResponse(BaseModel):
    """删除文档响应"""
    success: bool = True
    error: str | None = None

File: yuairun-frontend/src/types/knowledge.ts
/** 知识库相关类型定义 */

/** 知识库信息 */
export interface KnowledgeBase {
  id: string
  name: string
  description: string
  isSystem: boolean
  docCount: number
  chunkCount: number
  createdAt: string
  updatedAt: string
}

/** 知识库文档信息 */
export interface KnowledgeDocument {
  id: string
  kbId: string
  filename: string
  fileType: string
  fileSize: number
  pageCount: number | null
  charCount: number
  chunkCount: number
  status: 'pending' | 'processing' | 'ready' | 'failed'
  errorMsg: string | null
  createdAt: string
}

/** 创建知识库请求 */
export interface CreateKnowledgeBaseRequest {
  name: string
  description?: string
}

/** 知识库列表响应 */
export interface KnowledgeBaseListResponse {
  success: boolean
  data: KnowledgeBase[]
  error?: string
}

/** 文档列表响应 */
export interface KnowledgeDocumentListResponse {
  success: boolean
  data: KnowledgeDocument[]
  error?: string
}

/** 上传文档响应 */
export interface DocumentUploadResponse {
  success: boolean
  documentId?: string
  error?: string
}

/** 删除响应 */
export interface DeleteResponse {
  success: boolean
  error?: string
}

/** 
 * 搜索模式（已废弃）
 * @deprecated AI 现已自主决定检索策略，无需手动选择模式。
 * 请使用 searchMode: 'agentic'（默认值）。
 * 旧值 'search' | 'knowledge_base' | 'hybrid' 仍被后端接受但不再用于路由。
 */
export type SearchMode = 'search' | 'knowledge_base' | 'hybrid' | 'agentic'
